src/github.com/getlantern/balancer/strategy.go

```diff
@@ -0,0 +1,106 @@
+package balancer
+
+import (
+	"math/rand"
+)
+
+// Random strategy gives even chance to each dialer, act as a baseline to other
+// strategies.
+func Random(dialers []*dialer) dialerHeap {
+	return dialerHeap{dialers, func(i, j int) bool {
+		// we don't need good randomness, skip seeding
+		if rand.Intn(2) == 0 {
+			return false
+		}
+		return true
+	}}
+}
+
+// Sticky strategy always pick the dialer with largest consecutive success
+// count or the smallest consecutive failure count
+func Sticky(dialers []*dialer) dialerHeap {
+	return dialerHeap{dialers, func(i, j int) bool {
+		mi := dialers[i].metrics()
+		mj := dialers[j].metrics()
+		return (mi.consecSuccesses - mi.consecFailures) >
+			(mj.consecSuccesses - mj.consecFailures)
+	}}
+}
+
+// Fastest strategy always pick the dialer with lowest recent average connect time
+func Fastest(dialers []*dialer) dialerHeap {
+	return dialerHeap{dialers, func(i, j int) bool {
+		mi := dialers[i].metrics()
+		mj := dialers[j].metrics()
+		return mi.avgConnTime < mj.avgConnTime
```

Comment:
@fffw Aside from the network, we need to consider the behavior of the server itself.  If a server starts being overloaded, does that have a similar effect on latency and throughput or is one affected more than the other?

We had a recent email thread entitled "Proxying super slow at peak times".  I did some measurements from the affected server by downloading a small, medium and large file.  As server performance degraded, the effect was most noticeable on the large file, suggesting that bandwidth may be more affected than latency.

I think the best way to understand this is to run some tests.  To simulate a server being overloaded CPU-wise, we could just run a program on the server that consumes a lot of CPU without doing any real work.